ext/dapr-ext-workflow/dapr/ext/workflow/dapr_workflow_client.py

```diff
@@ -0,0 +1,212 @@
+# -*- coding: utf-8 -*-
+
+"""
+Copyright 2023 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+from __future__ import annotations
+from datetime import datetime
+from typing import Any, TypeVar, Union
+from dapr.conf import settings
+
+from durabletask import client
+from dapr.ext.workflow.workflow_state import WorkflowState
+from dapr.ext.workflow.workflow_context import Workflow
+
+T = TypeVar('T')
+TInput = TypeVar('TInput')
+TOutput = TypeVar('TOutput')
+
+
+class DaprWorkflowClient:
+    """Defines client operations for managing Dapr Workflow instances.
+
+       This is an alternative to the general purpose Dapr client. It uses a gRPC connection to send
+       commands directly to the workflow engine, bypassing the Dapr API layer.
+
+       This client is intended to be used by workflow applications, not by general purpose
+       applications.
+    """
+    def __init__(self, host: Union[str, None] = None, port: Union[str, None] = None):
+        if host is None:
+            host = settings.DAPR_RUNTIME_HOST
+        if not host or len(host) == 0 or len(host.strip()) == 0:
+            host = "localhost"
+        if port is None:
+            port = settings.DAPR_GRPC_PORT
+        if port is None:
+            raise ValueError("Port cannot be empty, please set DAPR_GRPC_PORT environment variable")
```

Comment:
If no port is specified, let's default to `50001`. This is the documented default port for gRPC.

Reply:
Actually, in `global_settings.py`, DAPR_GRPC_PORT is already defined to its default: https://github.com/dapr/python-sdk/blob/master/dapr/conf/global_settings.py#L24
So, it is just an extra check.

Reply:
Ah, my bad. It might actually be better to remove the check to avoid confusion. Too much defense in depth can be misleading.